collect_app/src/main/java/org/odk/collect/android/utilities/QRCodeUtils.java

```diff
@@ -36,48 +37,40 @@
 import com.google.zxing.qrcode.QRCodeWriter;
 import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
 
-import org.json.JSONException;
 import org.odk.collect.android.R;
 import org.odk.collect.android.application.Collect;
 
 import java.io.File;
-import java.io.FileOutputStream;
 import java.io.IOException;
+import java.security.MessageDigest;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.EnumMap;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.zip.DataFormatException;
 
+import io.reactivex.Observable;
 import timber.log.Timber;
 
-/**
- * Created by shobhit on 13/4/17.
- */
 
 public class QRCodeUtils {
+    public static final String QR_CODE_FILEPATH = Collect.SETTINGS + File.separator + "collect-settings.jpeg";
```

Comment:
Are we saving a QR Code image as a JPEG? Is JPEG the right compression format for this type of image. I should think PNG would be better.

Reply:
:+1:

Reply:
I wonder if these blurry edges are JPEG compression artifacts. Let’s look again after you change to PNG.

![screen shot 2018-02-26 at 10 02 15 pm](https://user-images.githubusercontent.com/55099/36713117-cc1815d4-1b40-11e8-890c-cbbdb34c1b7c.png)